deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c

```diff
@@ -55,9 +97,22 @@ GrB_Info GxB_MatrixTupleIter_iterate_row
 			rowIdx, iter->nrows) ;
 	}
 
-	iter->nvals = iter->A->p[rowIdx + 1];
-	iter->nnz_idx = iter->A->p[rowIdx];
+	GrB_Index rowIdx_sparse;
+	bool hypersparse_and_row_is_empty = false;
+	if(iter->sparsity_type == GxB_SPARSE) {
+		rowIdx_sparse = rowIdx;
+	} else {  // GxB_HYPERSPARSE - find the row_index in the sparse vector
+		if(!_find_row_index_in_Ah(iter, rowIdx, &rowIdx_sparse)) hypersparse_and_row_is_empty = true;
+	}
+
+	// If hypersparse and row is empty setting nvals = 0 cause iterator.next should return depleted
```

Comment:
Rephrase:
`// incase matrix is hyper-sparse and iterated row is empty, set 'nvals' to 0, this will cause the next call to 'next' to report depleted`